Gems/Atom/RPI/Code/Source/RPI.Reflect/Image/StreamingImageAssetHandler.cpp

```diff
@@ -36,14 +36,155 @@ namespace AZ
+                                {
+                                    // if the asset was loaded, trigger reload
+                                    foundAsset.Reload();
+                                    mipChainAsset.m_asset = foundAsset;
+                                }
+                                else
+                                {
+                                    // The asset wasn't loaded, queue load
+                                    // This may happen if the image's resolution increased which generate new mipchain assets 
+                                    mipChainAsset.m_asset.QueueLoad();
+                                }
+                                pendingAssetInfo.m_mipChainAssetSubIds.push_back(mipChainAssetId.m_subId);
+                                Data::AssetBus::MultiHandler::BusConnect(mipChainAssetId);
+                            }
+                        }
+
+                        if (pendingAssetInfo.m_mipChainAssetSubIds.size() > 0)
+                        {
+                            AZStd::scoped_lock<AZStd::mutex> lock(m_accessPendingAssetsMutex);
+                            m_pendingReloadImageAsset[imageGuid] = AZStd::move(pendingAssetInfo);
+                        }
+                    }
                 }
             }
 
             return loadResult;
         }
 
+        void StreamingImageAssetHandler::OnAssetReady(Data::Asset<Data::AssetData> asset)
+        {
+            HandleMipChainAssetLoad(asset, true);
+        }
+
+        void StreamingImageAssetHandler::OnAssetError(AZ::Data::Asset<AZ::Data::AssetData> asset)
+        {
+            HandleMipChainAssetLoad(asset, false);
+        }
+        
+        void StreamingImageAssetHandler::OnAssetReloaded(Data::Asset<Data::AssetData> asset)
+        {
+            HandleMipChainAssetLoad(asset, true);
+        }
+
+        void StreamingImageAssetHandler::OnAssetReloadError(Data::Asset<Data::AssetData> asset)
+        {
+            HandleMipChainAssetLoad(asset, false);
+        }
+        
+        void StreamingImageAssetHandler::HandleMipChainAssetBuses(Data::Asset<Data::AssetData> streamingImageAsset, bool connect)
+        {
+            StreamingImageAsset* assetData = streamingImageAsset.GetAs<StreamingImageAsset>();
+            for (auto& mipChainAsset : assetData->m_mipChains)
+            {
+                // Need to skip tail mipchain which is in the list and doesn't have an asset id.
+                if (mipChainAsset.m_asset.GetId().IsValid())
+                {
+                    if (connect)
+                    {
+                        Data::AssetBus::MultiHandler::BusConnect(mipChainAsset.m_asset.GetId());
+                    }
+                    else
+                    {
+                        Data::AssetBus::MultiHandler::BusDisconnect(mipChainAsset.m_asset.GetId());
```

Comment:
Suggestion: Might want to add a BusDisconnect() into the destructor to make sure it doesn't leave any dangling connections from timing issues.